### PR TITLE
feat(ecmascript): Unify InternalSlots and InternalMethods trait default implementations to remove redundancy

### PR DESCRIPTION
**Pull request by aapoalas**
@andreubotella had an excellent point earlier this week at WEHF: We _can_ reuse the InternalMethods implementations if we just make some minor changes.

This refactoring also made me realize that we can think of the "backing object" as a custom internal slot. This may or may not be a helpful viewpoint for thinking about how the engine corresponds to the spec.

### CODE REVIEW

**Comment by MierenManz** [comment]
I'll try to review when I'm on pc. @aapoalas want to walk me through the pr on discord tomorrow evening? I haven't really followed the github compared to the discord

**Comment by aapoalas** [comment]
> I'll try to review when I'm on pc. @aapoalas want to walk me through the pr on discord tomorrow evening? I haven't really followed the github compared to the discord

Yeah, sure thing.

**Comment by andreubotella** [comment]
I'll try to review tomorrow evening, won't really have time before that.

File: nova_vm/src/ecmascript/types/language/object/internal_methods.rs [hunk 1]
```diff
@@ -1,56 +1,183 @@
...
+    ) -> JsResult<Value> {
+        // 1. Return ? OrdinaryGet(O, P, Receiver).
+        match self.get_backing_object(agent) {
+            Some(backing_object) => {
+                ordinary_get(agent, backing_object.into_object(), property_key, receiver)
+            }
+            None => {
+                // a. Let parent be ? O.[[GetPrototypeOf]]().
+                let Some(parent) = self.internal_get_prototype_of(agent)? else {
+                    return Ok(Value::Undefined);
```
Comment (sno2) [inline nova_vm/src/ecmascript/types/language/object/internal_methods.rs:173]:
```suggestion
                    // b. If parent is null, return undefined.
                    return Ok(Value::Undefined);
```

**Review by sno2 — DISMISSED** [review]
One comment nitpick. Very nice work and the ratio on this diff proves this is the right idea in my eyes👏

**Review by MierenManz — APPROVED** [review]
LGBT

**Comment by MierenManz** [comment]
> LGBT

I mean LGTM! :sweat_smile:

**Pull request merged**